### PR TITLE
[PAY-922] use correct k2 values in t3

### PR DESCRIPTION
**Pull request by mattravenhill-humi**
https://gethumi.atlassian.net/browse/PAY-922

### CODE REVIEW

File: lib/taxman/taxman2023/tax_calculator.rb [hunk 1]
```diff
@@ -70,13 +54,43 @@ def calculate
...
+          puts "K2RQ"
+          context[:k2_value] = K2RQ.amount(context)
+        else
+          puts "K2Q"
+          context[:k2_value] = K2Q.amount(context)
+        end
       else
         calculate_provincial_non_qc_taxes
 
         # Set provincial tax amount
```
Comment (claus-humi) [inline lib/taxman/taxman2023/tax_calculator.rb:69]: I think I would like to keep the order of first calculating federal taxes (T3) and then provincial, so I would extract setting the K2 values into a method that is called after setting T1, but before branching on what taxes to calculate.

**Review by claus-humi — COMMENTED** [review]
Basics looks good, just a bit of ordering.

File: lib/taxman/taxman2023/tax_calculator.rb (hunk 1 above)
Comment (mattravenhill-humi) [reply lib/taxman/taxman2023/tax_calculator.rb]: Alright, I've moved the T3 code back into place and created a new method for calculating the K2 value.

File: lib/taxman/taxman2023/tax_calculator.rb [hunk 2]
```diff
@@ -127,6 +125,22 @@ def check_required_context
...
+                               K2RQ.amount(context)
+                             else
+                               K2Q.amount(context)
+                             end
+                           elsif context[:previously_on_qpp]
+                             # set Non Quebec k2 values
+                             K2R.amount(context)
+                           else
+                             K2.amount(context)
+                           end
```
Comment (claus-humi) [inline lib/taxman/taxman2023/tax_calculator.rb:141]: I know this is "the ruby way" but it always looks terrible to me lol. I guess my heart is still made of gophers and not gems! Not a change request, just a comment!

**Review by claus-humi — APPROVED** [review]

File: lib/taxman/taxman2023/tax_calculator.rb (hunk 2 above)
Comment (mattravenhill-humi) [reply lib/taxman/taxman2023/tax_calculator.rb:141]: I agree that it looks a little wonky, this change was made by Rubocop.

**Pull request merged**